compiler/src/test/java/dagger/internal/codegen/GraphValidationTest.java

```diff
@@ -2131,6 +2268,285 @@ public void subcomponentBindingConflictsWithParentWithNullableViolationAsWarning
+            "",
+            GENERATED_ANNOTATION,
+            "public final class DaggerParent implements Parent {",
+            "  private MembersInjector<Injected> injectedMembersInjector;",
+            "",
+            "  private DaggerParent(Builder builder) {",
+            "    assert builder != null;",
+            "    initialize(builder);",
+            "  }",
+            "",
+            "  public static Builder builder() {",
+            "    return new Builder()",
+            "  }",
+            "",
+            "  public static Parent create() {",
+            "    return new Builder().build();",
+            "  }",
+            "",
+            "  @SuppressWarnings(\"unchecked\")",
+            "  private void initialize(final Builder builder) {",
+            "    this.injectedMembersInjector = Injected_MembersInjector",
+            "        .create(Parent_ParentModule_ObjectFactory.create());",
+            "  }",
+            "",
+            "  @Override",
+            "  public void inject(Injected injected) {",
+            "    injectedMembersInjector.injectMembers(injected);",
+            "  }",
+            "",
+            "  @Override",
+            "  public Child child() {",
+            "    return new ChildImpl();",
+            "  }",
+            "",
+            "  public static final class Builder {",
+            "    private Builder() {}",
+            "",
+            "    public Parent build() {",
+            "      return new DaggerParent(this);",
+            "    }",
+            "",
+            "    @Deprecated",
+            "    public Builder parentModule(Parent.ParentModule parentModule) {",
+            "      Preconditions.checkNotNull(parentModule);",
+            "      return this;",
+            "    }",
+            "  }",
+            "",
+            "  private final class ChildImpl implements Child {",
+            "    private MembersInjector<Injected> injectedMembersInjector;",
+            "",
+            "    private ChildImpl() {",
+            "      initialize();",
+            "    }",
+            "",
+            "    @SuppressWarnings(\"unchecked\")",
+            "    private void initialize() {",
+            "      this.injectedMembersInjector = Injected_MembersInjector",
+            "          .create(Parent_ParentModule_ObjectFactory.create());",
+            "    }",
+            "",
+            "    @Override",
+            "    public void inject(Injected injected) {",
+            "      injectedMembersInjector.injectMembers(injected);",
+            "    }",
+            "  }",
+            "}");
+    assertAbout(javaSources())
+        .that(ImmutableList.of(parent, child, injected))
+        .processedWith(new ComponentProcessor())
+        .compilesWithoutError()
+        .and()
+        .generatesSources(generatedComponent);
+  }
+
+  @Test
+  public void subcomponentGeneratedCorrectlyWhenBothComponentsPerformMultiboundMembersInjection() {
```

Comment:
This should probably be a functional test, since what we care about is that injection from the child gives a different value than from the parent. The previous test should be kept, because we do care about the internals of the code (i.e. that a MembersInjector isn't recreated